### PR TITLE
Merge m3collector into monorepo

### PR DESCRIPTION
**Pull request by robskillington**

### CODE REVIEW

File: src/collector/glide.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+package: github.com/m3db/m3collector
```
Comment (prateek) [inline src/collector/glide.yaml]: can you migrate deps to central glide and delete this and glide.lock in the subfolder

File: src/collector/LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
+                                 Apache License
```
Comment (prateek) [inline src/collector/LICENSE]: can delete this

File: src/collector/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+Contributing
```
Comment (prateek) [inline src/collector/CONTRIBUTING.md]: this too

File: src/collector/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+Changelog
```
Comment (prateek) [inline src/collector/CHANGELOG.md]: this too

File: src/collector/.travis.yml [hunk 5]
```diff
@@ -0,0 +1,16 @@
+language: go
```
Comment (prateek) [inline src/collector/.travis.yml]: this too

File: src/collector/.metalinter.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (prateek) [inline src/collector/.metalinter.json]: this too

File: src/collector/.gitignore [hunk 7]
```diff
@@ -0,0 +1,24 @@
+*.out
```
Comment (prateek) [inline src/collector/.gitignore]: this too

File: src/collector/.excludemetalint [hunk 8]
```diff
@@ -0,0 +1,3 @@
+vendor/
```
Comment (prateek) [inline src/collector/.excludemetalint]: this too

File: Makefile [hunk 9]
```diff
@@ -49,6 +49,7 @@ SERVICES :=     \
 SUBDIRS :=    \
 	x           \
 	cmd         \
+	collector   \
```
Comment (prateek) [inline Makefile]: interesting, no new services in the subrepo?

File: src/collector/Makefile [hunk 10]
```diff
@@ -0,0 +1,114 @@
+SELF_DIR := $(dir $(lastword $(MAKEFILE_LIST)))
```
Comment (prateek) [inline src/collector/Makefile]: delete this one too

**Comment by prateek** [comment]
just in case we forget, same reminder re: squashing new changes into single commit and merging the rest w/o squashing

File: Makefile (hunk 9 above)
Comment (robskillington) [reply Makefile]: Nope, that's coming with my new changes heh.

Comment (prateek) [reply Makefile]: Ah cheers

**Pull request merged**